tools/osv-scanner/cmd/osv-scanner/main_test.go

```diff
@@ -0,0 +1,186 @@
+//nolint:testpackage // main cannot be accessed directly, so cannot use main_test
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func dedent(t *testing.T, str string) string {
+	t.Helper()
+
+	// 0. replace all tabs with spaces
+	str = strings.ReplaceAll(str, "\t", "  ")
+
+	// 1. remove trailing whitespace
+	re := regexp.MustCompile(`\r?\n([\t ]*)$`)
+	str = re.ReplaceAllString(str, "")
+
+	// 2. if any of the lines are not indented, return as we're already dedent-ed
+	re = regexp.MustCompile(`(^|\r?\n)[^\t \n]`)
+	if re.MatchString(str) {
+		return str
+	}
+
+	// 3. find all line breaks to determine the highest common indentation level
+	re = regexp.MustCompile(`\n[\t ]+`)
+	matches := re.FindAllString(str, -1)
+
+	// 4. remove the common indentation from all strings
+	if matches != nil {
+		size := len(matches[0]) - 1
+
+		for _, match := range matches {
+			if len(match)-1 < size {
+				size = len(match) - 1
+			}
+		}
+
+		re := regexp.MustCompile(`\n[\t ]{` + fmt.Sprint(size) + `}`)
+		str = re.ReplaceAllString(str, "\n")
+	}
+
+	// 5. Remove leading whitespace.
+	re = regexp.MustCompile(`^\r?\n`)
+	str = re.ReplaceAllString(str, "")
+
+	return str
+}
+
+// checks if two strings are equal, treating any occurrences of `%%` in the
+// expected string to mean "any text"
+func areEqual(t *testing.T, actual, expect string) bool {
+	t.Helper()
+
+	expect = regexp.QuoteMeta(expect)
+	expect = strings.ReplaceAll(expect, "%%", ".+")
+
+	re := regexp.MustCompile(`^` + expect + `$`)
+
+	return re.MatchString(actual)
+}
+
+type cliTestCase struct {
+	name         string
+	args         []string
+	wantExitCode int
+	wantStdout   string
+	wantStderr   string
+}
+
+func testCli(t *testing.T, tc cliTestCase) {
```

Comment:
nit: this should be testCLI per https://github.com/golang/go/wiki/CodeReviewComments#initialisms